### PR TITLE
Main menu

### PR DESCRIPTION
**Pull request by LuukSantegoeds**
# Description

We now have a fancy main menu with a logo and a button to "start the game"

## Screenshot(s)
image

# How can this be tested?
You can start the game and go to the menu by clicking the button.

- [ ] Can launch the game
- [ ] Can navigate to the level

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] Any dependent changes have been merged and published in downstream modules
- [X] Included 1 or more screenshots
- [X] Code is const correct

### CODE REVIEW

File: include/menu/button.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#include "menu/text.hpp"
+
+struct Button {
+public:
+    std::string texture_path;
+    int alpha;
+
+    int x; // Before relative modifier
+    int y; // Before relative modifier
+    int xScale; // Before relative modifier
```
Comment (Bram-Boris) [inline include/menu/button.hpp]: snake_case.

File: include/menu/image.hpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#include <string>
+
+struct Image {
+public:
+    std::string texture_path;
+    int alpha;
+
+    int x; // Before relative modifier
+    int y; // Before relative modifier
+    int xScale; // Before relative modifier
```
Comment (Bram-Boris) [inline include/menu/image.hpp]: snake_case.

File: include/menu/text.hpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+struct Text {
+public:
+    std::string text;
+    int font_size;
+    Color color;
+
+    int x; // Before relative modifier
+    int y; // Before relative modifier
+    int xScale; // Before relative modifier
```
Comment (Bram-Boris) [inline include/menu/text.hpp]: snake_case.

File: lib/entities/entity_factory.cpp [hunk 4]
```diff
@@ -56,7 +53,7 @@ int EntityFactory::createWeapon(double xPos, double yPos) const{
     comps->push_back(std::make_unique<PhysicsComponent>(50, 0, 0, 0, true, Kinematic::IS_NOT_KINEMATIC, true, true));
     comps->push_back(std::make_unique<TextureComponent>(std::move(r)));
     comps->push_back(std::make_unique<PickupComponent>());
-    //comps->push_back(std::make_unique<WeaponComponent>(1, 1, 10, 5));
+    // comps->push_back(std::make_unique<WeaponComponent>(1, 1, 10, 5));
```
Comment (Bram-Boris) [inline lib/entities/entity_factory.cpp]: ?

File: lib/scenes/scene.cpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    this->bg_music = json.getString("bg_music");
+}
+
+Scene::Scene(int screen_width, int screen_height, int width, int height) {
+    calculateRelativeModifier(screen_width, screen_height, width, height);
+}
+
+void Scene::calculateRelativeModifier(int screen_width, int screen_height, int width, int height) {
+    double relative_modifier = width / (double)screen_width;
+    if(!height / screen_height == relative_modifier) {
```
Comment (Bram-Boris) [inline lib/scenes/scene.cpp]: You are casting to a double but then comparing it to an int? both or none.

**Review by Bram-Boris — COMMENTED** [review]
Missing some snakes.

File: lib/entities/entity_factory.cpp (hunk 4 above)
Comment (MarkMeer) [reply lib/entities/entity_factory.cpp]: Not intentional, Luuk quoted: "Een fout van mij". This will be reverted in the next commit.

**Review by Bram-Boris — APPROVED** [review]
So it didn't even work I guess(relative modifier). Anyway it looks slick! Nice job

**Comment by MarkMeer** [comment]
> So it didn't even work I guess(relative modifier). Anyway it looks slick! Nice job

Yeah, I myself did not notice this until I tried applying your feedback to the method where it is used. Goes to show that reviews are a very good thing 👍 .

**Review by Peter-JanGootzen — CHANGES_REQUESTED** [review]
The EntityFactory should not be used as a single point of entry when creating entities. The way structs are used for creating buttons and images is really convoluted, just put a method in Menu that creates all the necessary entities/components for the menu. You could do this via the EntityFactory or the Entitymanager.

EDIT:
While looking through the PR again and discussing it with Bram and Jan. We found out that Level also works like this. Eventhough fixing this would create two different approaches to creating entities in the SceneManager, we would still like to see this changed.

**Comment by Bram-Boris** [comment]
Due to time limitations we have to merge this PR even though the technical side is not really what we want.

**Pull request merged**